pkg/pool-utils/test/BasePoolController.test.ts

```diff
@@ -0,0 +1,236 @@
+  });
+
+  allTokens = await TokenList.create(['MKR', 'DAI', 'SNX', 'BAT'], { sorted: true });
+  await allTokens.mint({ to: admin, amount: fp(100) });
+});
+
+describe('BasePoolController', function () {
+  const NEW_SWAP_FEE = fp(0.05);
+  const NEXT_SWAP_FEE = fp(0.005);
+
+  sharedBeforeEach('deploy controller and pool', async () => {
+    poolController = await deploy('BasePoolController');
+
+    const params = {
+      vault,
+      tokens: allTokens,
+      weights: WEIGHTS,
+      owner: poolController.address,
+      swapFeePercentage: POOL_SWAP_FEE_PERCENTAGE,
+      poolType: WeightedPoolType.WEIGHTED_POOL,
+      swapEnabledOnStart: false,
+    };
+    pool = await WeightedPool.create(params);
+  });
+
+  context('pool controller not initialized', () => {
+    it('creates the pool', async () => {
+      expect(await pool.getOwner()).to.equal(poolController.address);
+    });
+
+    it('sets up the pool controller', async () => {
+      expect(await poolController.owner()).to.equal(admin.address);
+    });
+
+    it('cannot call functions before initialization', async () => {
+      await expect(poolController.connect(admin).setSwapFeePercentage(NEW_SWAP_FEE)).to.be.revertedWith(
+        'UNINITIALIZED'
+      );
+    });
+
+    it('has no initial metadata', async () => {
+      expect(await poolController.getMetadata()).to.equal('0x');
+    });
+
+    it('owner can set metadata', async () => {
+      const tx = await poolController.connect(admin).updateMetadata(METADATA);
+      expectEvent.inReceipt(await tx.wait(), 'MetadataUpdated', {
+        metadata: METADATA,
+      });
+
+      expect(await poolController.getMetadata()).to.equal(METADATA);
+    });
+
+    it('non-owner cannot set metadata', async () => {
+      await expect(poolController.connect(other).updateMetadata(METADATA)).to.be.revertedWith('CALLER_IS_NOT_OWNER');
+    });
+  });
+
+  context('pool controller is initialized', () => {
+    sharedBeforeEach('initialize pool controller', async () => {
+      await poolController.bindPool(pool.address);
+    });
+
+    it('owner can set the swap fee', async () => {
+      await poolController.connect(admin).setSwapFeePercentage(NEW_SWAP_FEE);
+
+      expect(await pool.getSwapFeePercentage()).to.equal(NEW_SWAP_FEE);
+    });
+
+    it('enforces owner permissions', async () => {
+      await expect(poolController.connect(other).setSwapFeePercentage(NEW_SWAP_FEE)).to.be.revertedWith(
+        'CALLER_IS_NOT_OWNER'
+      );
+    });
+
+    it('can be transferred', async () => {
+      await poolController.transferOwnership(other.address);
+
+      await expect(poolController.connect(admin).setSwapFeePercentage(NEXT_SWAP_FEE)).to.be.revertedWith(
+        'CALLER_IS_NOT_OWNER'
+      );
+
+      await poolController.connect(other).setSwapFeePercentage(NEXT_SWAP_FEE);
+
+      expect(await pool.getSwapFeePercentage()).to.equal(NEXT_SWAP_FEE);
+    });
+
+    it('can renounce ownership', async () => {
```

Comment:
This feels overkill, we're essentially testing that `Ownable` works as expected, which is a given. What we should test instead is that we use `Ownable` correctly (i.e. we apply the `onlyOwner` modifier).

Reply:
Ok; above tests cover that, so I can just remove this.